dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java

```diff
@@ -226,123 +233,92 @@ public static HashMap<String, List<String>> importFile(
+     * error tracking, and processing information.
+     *
+     * <p>The preview parameter determines the behavior:
+     * <ul>
+     *   <li>{@code preview == true}: Analyzes and validates data before committing changes</li>
+     *   <li>{@code preview == false}: Attempts immediate import with result summary</li>
+     * </ul>
+     *
+     * @param importId                 ID of this data import
+     * @param currentSiteId            ID of the Site for content
+     * @param contentTypeInode         Content Type Inode
+     * @param keyfields                Field Inodes for content matching
+     * @param preview                  If true, validates without saving
+     * @param isMultilingual           If true, handles multiple languages
+     * @param user                     User performing the import
+     * @param language                 Language ID (-1 for language from CSV)
+     * @param csvHeaders               Headers from CSV file
+     * @param csvreader                CSV file reader
+     * @param languageCodeHeaderColumn Language code column index
+     * @param countryCodeHeaderColumn  Country code column index
+     * @param wfActionId               Workflow action ID
+     * @param fileTotalLines           The total lines count of the CSV file
+     * @param request                  HTTP request context
+     * @param progressCallback         Optional callback for progress updates
+     * @return Structured ImportResult containing validation and processing details
+     * @throws DotRuntimeException If a critical error occurs
+     * @throws DotDataException    If a data access error occurs
+     */
+    public static ImportResult importFileResult(
+            Long importId, String currentSiteId, String contentTypeInode, String[] keyfields,
+            boolean preview, boolean isMultilingual, User user, long language,
+            String[] csvHeaders, CsvReader csvreader, int languageCodeHeaderColumn,
+            int countryCodeHeaderColumn, String wfActionId, final long fileTotalLines,
             final HttpServletRequest request, final LongConsumer progressCallback)
             throws DotRuntimeException, DotDataException {
 
-        final HashMap<String, List<String>> results = new HashMap<>();
-        results.put(KEY_WARNINGS, new ArrayList<>());
-        results.put(KEY_ERRORS, new ArrayList<>());
-        results.put(KEY_MESSAGES, new ArrayList<>());
-        results.put(KEY_RESULTS, new ArrayList<>());
-        results.put(KEY_COUNTERS, new ArrayList<>());
-        results.put(KEY_IDENTIFIERS, new ArrayList<>());
-        results.put(KEY_UPDATED_INODES, new ArrayList<>());
-        results.put(KEY_LAST_INODE, new ArrayList<>());
-        results.put(Contentlet.WORKFLOW_ACTION_KEY, new ArrayList<>());
-
-        Structure contentType = CacheLocator.getContentTypeCache().getStructureByInode (contentTypeInode);
+        Structure contentType = CacheLocator.getContentTypeCache()
```

Comment:
why using legacy beans?

Reply:
One of the requirements for the ticket was to leave the logic exactly as it currently is.